lib/noticed/delivery_methods/base.rb

```diff
@@ -4,8 +4,31 @@ class Base < Noticed.parent_class.constantize
       extend ActiveModel::Callbacks
       define_model_callbacks :deliver
 
+      class_attribute :option_names, instance_writer: false, default: []
+
       attr_reader :notification, :options, :params, :recipient, :record
 
+      class << self
+        # Copy option names from parent
+        def inherited(base) #:nodoc:
+          base.option_names = option_names.dup
+          super
+        end
+
+        def options(*names)
+          option_names.concat Array.wrap(names)
+        end
+        alias option options
+
+        def validate!(delivery_method_options)
+          option_names.each do |option_name|
+            if delivery_method_options[option_name].nil?
```

Comment:
Question: I am wondering if we should avoid going to the length of making this akin to `validates_presence_of` validator like in Rails. In the previous implementation, we only check if the `key` is present with this specific name, and not the value of the item itself.

I wonder if checking if the value itself is `nil` could be troublesome at least to some users of the gem, in case there is a legitimate requirement to pass `nil` as the value of an option or if the value is set by an ActiveRecord query which may return `nil` in some cases, say like `also_send_to: User.find_by(id: admin_id)`.

Or is it deliberate that you wanted to make this a presence validator? Would like to know your thoughts 🙂

Reply:
You are totally right, this would break backwards compatibility. The reason of checking for the options presence, is that it is the same validation used for notifications [params](https://github.com/excid3/noticed/blob/ed0527763c3bf1a326054ac98080716136ca5396/lib/noticed/base.rb#L121).
I agree with you that it will make more sense to only check if the key is present in the options hash. But before making the change, I'd like to know @excid3 opinion on it, because we may also want to apply this change to the params validation, so everything is consistent.

Reply:
I think it should only check for the presence of the option, not the value. We don't really care what the value is because that's up to the delivery method to decide, so long as something was passed in.